src/compat/posix/pthread/pthread_barrier.c

```diff
@@ -0,0 +1,207 @@
+/**
+ * @file   
+ * @brief  pthread_barrier functions
+ * @date   11.03.2020
+ * @author Sobhan Mondal <inro20fdm@gmail.com>
+ */
+
+#include <stdlib.h>
+#include <errno.h>
+#include <pthread.h>
+#include <semaphore.h>
+#include <util/err.h>
+
+static void sem_post_multiple(sem_t *sem, int count) { 
+    for (; count > 0; --count) {
+        sem_post(sem);
+    }
+} 
+
+/*
+ * The pthread_barrier_init() function shall fail if:
+ *
+ * [EBUSY]
+ *	   The implementation has detected an attempt to 
+ *     reinitialize a barrier while it is in use.
+ * [EINVAL]
+ *     The value specified by count is equal to zero 
+ *	   or the barrier is not defined.
+ * [ENOMEM]
+ *     Insufficient memory exists to initialize the barrier.
+ *
+ * On success, The pthread_barrier_init() function will return 0.
+ * The pthread_barrier_init() function shall not return an 
+ * error code of [EINTR].
+ */
+
+int pthread_barrier_init(pthread_barrier_t *barrier, 
+	const pthread_barrierattr_t *attr, unsigned int count) {	
+	pthread_barrier_t ib;
+
+	if (NULL == barrier || 0 == count) {
+		return EINVAL;
+	}
+
+	if (NULL != (*barrier)) {
+        if (1 == (*barrier)->initialize_flag) {
+            return EBUSY;
+        }
+    }
```

Comment:
I see mix of spaces and tab here. Please use tabs :)

Reply:
Done

Reply:
Done.